conda-store-server/tests/plugins/lock/test_conda_lock.py

```diff
@@ -0,0 +1,110 @@
+# Copyright (c) conda-store development team. All rights reserved.
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
+
+from unittest import mock
+
+import pytest
+import yaml
+
+from conda_store_server._internal import conda_utils
+from conda_store_server.plugins import plugin_context
+from conda_store_server.plugins.lock import conda_lock
+
+
+@pytest.mark.parametrize(
+    "specification",
+    [
+        "simple_specification",
+        "simple_specification_with_pip",
+    ],
+)
+@mock.patch("conda_store_server.plugins.lock.conda_lock.run_lock")
+@pytest.mark.long_running_test
+def test_solve_lockfile(
+    mock_run_lock,
+    specification,
+    request,
+):
+    """Test that the call to conda_lock.run_lock is formed correctly."""
+
+    # Dump dummy data to the expected lockfile output location
+    def run_lock_side_effect(lockfile_path, **kwargs):
+        with open(lockfile_path, "w") as f:
+            yaml.dump({"foo": "bar"}, f)
+
+    mock_run_lock.side_effect = run_lock_side_effect
+
+    platforms = [conda_utils.conda_platform()]
+    specification = request.getfixturevalue(specification)
+    if specification.variables is None:
+        cuda_version = None
+    else:
+        cuda_version = specification.variables.get("CONDA_OVERRIDE_CUDA")
+
+    locker = conda_lock.CondaLock(conda_command="mamba")
+    lock_result = locker.lock_environment(
+        context=plugin_context.PluginContext(),
+        spec=specification,
+        platforms=platforms,
+    )
+
+    # Check that the call to `conda_lock` is correctly formed
+    mock_run_lock.assert_called_once()
+    call_args = mock_run_lock.call_args_list[0][1]
+    assert str(call_args["environment_files"][0]).endswith("environment.yaml")
+    assert call_args["platforms"] == platforms
+    assert str(call_args["lockfile_path"]).endswith("conda-lock.yaml")
+    assert call_args["conda_exe"] == "mamba"
+    assert call_args["with_cuda"] == cuda_version
+
+    assert lock_result["foo"] == "bar"
+
+
+def test_solve_lockfile_simple(simple_specification):
+    locker = conda_lock.CondaLock(
+        conda_command="mamba", conda_flags="--strict-channel-priority"
+    )
+    lock_result = locker.lock_environment(
+        context=plugin_context.PluginContext(),
+        spec=simple_specification,
+        platforms=[conda_utils.conda_platform()],
+    )
+    assert len(lock_result["package"]) != 0
```

Comment:
Shall we check for some minimum contents; e.g. that a certain package name is present?

Reply:
Ah sorry, I see that this module was copied from a different location. Anyway the comments still apply if you have the time / energy :)